lightning/src/io/mod.rs

```diff
@@ -0,0 +1,172 @@
+pub use bitcoin::io::*;
+
+/// The `Seek` trait provides a cursor which can be moved within a stream of
+/// bytes.
+///
+/// The stream typically has a fixed size, allowing seeking relative to either
+/// end or the current offset.
+pub trait Seek {
+	/// Seek to an offset, in bytes, in a stream.
+	///
+	/// A seek beyond the end of a stream is allowed, but behavior is defined
+	/// by the implementation.
+	///
+	/// If the seek operation completed successfully,
+	/// this method returns the new position from the start of the stream.
+	/// That position can be used later with [`SeekFrom::Start`].
+	///
+	/// # Errors
+	///
+	/// Seeking to a negative offset is considered an error.
+	fn seek(&mut self, pos: SeekFrom) -> Result<u64>;
+}
+
+/// Enumeration of possible methods to seek within an I/O object.
+///
+/// It is used by the [`Seek`] trait.
+#[derive(Copy, PartialEq, Eq, Clone, Debug)]
+pub enum SeekFrom {
+	/// Sets the offset to the provided number of bytes.
+	Start(u64),
+
+	/// Sets the offset to the size of this object plus the specified number of
+	/// bytes.
+	///
+	/// It is possible to seek beyond the end of an object, but it's an error to
+	/// seek before byte 0.
+	End(i64),
+
+	/// Sets the offset to the current position plus the specified number of
+	/// bytes.
+	///
+	/// It is possible to seek beyond the end of an object, but it's an error to
+	/// seek before byte 0.
+	Current(i64),
+}
+
+/// Adaptor to chain together two readers.
+pub struct Chain<T, U> {
+	first: T,
+	second: U,
+	done_first: bool,
+}
+
+impl<T: Read, U: Read> Read for Chain<T, U> {
+	fn read(&mut self, buf: &mut [u8]) -> Result<usize> {
+		if !self.done_first {
+			match self.first.read(buf)? {
+				0 if !buf.is_empty() => self.done_first = true,
+				n => return Ok(n),
+			}
+		}
+		self.second.read(buf)
+	}
+}
+
+/// Emulation of std::io::Cursor
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct Cursor<T> {
+	inner: T,
+	pos: u64,
+}
```

Comment:
Why are you reimplementing this? It's already in `bitcoin-io`.

Reply:
Oh, the type is broken, you could've reported it. Please report important missing APIs next time. I did it for you now: https://github.com/rust-bitcoin/rust-bitcoin/issues/3174